strmangle.go

```diff
@@ -237,6 +237,10 @@ var (
 // Note: This method is ugly because it has been highly optimized,
 // we found that it was a fairly large bottleneck when we were using regexp.
 func TitleCase(n string) string {
+	// Replace colons and dots with underscore so the string can be correctly uppercased.
+	replacer := strings.NewReplacer(":", "_", ".", "_")
+	n = replacer.Replace(n)
```

Comment:
I think any such manipulation should come after the attempt to fetch from the cache.